hangman.py

```diff
@@ -83,63 +77,170 @@ def get_available_letters(letters_guessed):
+    print('-----------------')
+    return warnings
+
+
+def check_guesses(guesses, user_guess, repeated_guesses, show_word):
+    '''
+
+    :param guesses: guesses that are given to user
+    :param user_guess: user's input
+    :param repeated_guesses: letters that have been repeated
+    :param show_word: output that is visible to user
+    :return: warnings that are left
+    '''
+    guesses -= 1
+    if not user_guess.isalpha():
+        print("Oops! That is not a valid letter. You have no warnings left so you lose one guess. " + show_word)
+    elif user_guess in repeated_guesses:
+        print(
+            'Oops! You\'ve already guessed that letter. You have no warnings left so you lose one guess: ' + show_word)
+    else:
+        print('Oops! That letter is not in my word: ' + show_word)
+    print('--------')
+    return guesses
+
 
 def hangman(secret_word):
     '''
     secret_word: string, the secret word to guess.
-    
+
     Starts up an interactive game of Hangman.
-    
-    * At the start of the game, let the user know how many 
+
+    * At the start of the game, let the user know how many
       letters the secret_word contains and how many guesses s/he starts with.
-      
+
     * The user should start with 6 guesses
 
     * Before each round, you should display to the user how many guesses
       s/he has left and the letters that the user has not yet guessed.
-    
+
     * Ask the user to supply one guess per round. Remember to make
       sure that the user puts in a letter!
-    
-    * The user should receive feedback immediately after each guess 
+
+    * The user should receive feedback immediately after each guess
       about whether their guess appears in the computer's word.
 
-    * After each guess, you should display to the user the 
+    * After each guess, you should display to the user the
       partially guessed word so far.
-    
+
     Follows the other limitations detailed in the problem write-up.
     '''
-    # FILL IN YOUR CODE HERE AND DELETE "pass"
-    pass
-
-
-
-# When you've completed your hangman function, scroll down to the bottom
-# of the file and uncomment the first two lines to test
-#(hint: you might want to pick your own
-# secret_word while you're doing your own testing)
-
-
-# -----------------------------------
-
+    letters_guessed = []
+    guesses = 0
+    warnings = 0
+    unique_letters = ''
+    for letter in secret_word:
+        if letter not in unique_letters:
+            unique_letters += letter
+
+    print('Welcome to the game Hangman!')
+    print('I\'m thinking of a word that is ' + str(len(secret_word)) + ' letters long.')
+
+    while guesses < 6:
+        letters_left = get_available_letters(letters_guessed)
+        print('------------')
+        print('You have ' + str(6 - guesses) + ' guesses left.')
+        print('Available letters: ' + letters_left)
+
+        user_guess = (input('Please guess a letter: '))
+
+        if str.isalpha(user_guess):
```

Comment:
`str` is a keyword in Python. Rename the variable. Correct using of string methods: `<name_of_str_var>.<name_of_method>() `